Vaccel 0.23 extend

demetresalx opened the pull request:
## Reason for This PR

Fixes #8 

## Description of Changes
-Automates the construction of the docker container images for x86_64 & aarch64 for changes in the Dockerfiles only.
-Pushes both archs under the same label via docker manifest
-Automates firecracker binaries building & uploading for pull requests (both x86 & aarch64) that will be needed for vaccel

demetresalx committed:
Add workflow to automate Docker image creation

Add a new github actions workflow for creating and publishing our own
Docker images for building Firecracker with vaccelrt support.

bchalios committed:
Clone recursively vaccelrt in Dockerfiles

vaccelrt has now submodules which we need to fetch before building the
project properly.

Signed-off-by: Babis Chalios <mail@bchalios.io>

demetresalx committed:
Strategy matrix for different archs

bchalios committed:
Fix cargo-audit version

The version that was being fetched, it was using nightly features
causing build to fail.

Signed-off-by: Babis Chalios <mail@bchalios.io>

bchalios committed:
Use multi-arch image for in Devtool

We created a multi-arch for building vAccel-enabled Firecracker. Use it
in devtool to make it simpler. At the moment, the multi-arch image is
being created by hand.

Signed-off-by: Babis Chalios <mail@bchalios.io>

demetresalx committed:
Docker manifest a multi-arch image

login to dockerhub and use docker manifest create to build & push
the multi-arch image devtool script needs to
produce the corresponding firecracker binaries

demetresalx committed:
Use custom upload action

Instead of the pedantic route to upload to s3 followed before

demetresalx committed:
Solely upload firecracker binary

Instead of the entire binary folder that we tried to upload before

bchalios commented on .github/workflows/pull_request_v0.23.yaml:
```diff
@@ -17,32 +22,13 @@ jobs:
+          url: https://s3.nubificus.co.uk
+          access-key: ${{ secrets.AWS_ACCESS_KEY }}
+          secret-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          local-path: build/cargo_target/${{ matrix.arch }}-unknown-linux-gnu/${{ matrix.build_type }}/firecracker
+          remote-path: nbfc-assets/github/firecracker/${{ matrix.arch }}/${{ matrix.build_type}}/
```
Here we should include in the remote path the SHA of the commit

bchalios commented:
Apart from the comment above the only thing remaining from this PR is squashing the commits in a set of meaningful commits.

demetresalx committed:
Add pull request SHA in artifact pathing

demetresalx committed:
Firecracker upload workflow triggered only via PR

demetresalx committed:
Docker image CI runs on jetson runner

due to docker manifest not being up to date in x86 runners,
one of the two aarch64 machines will be used.
